omnibus/package-scripts/agent/preinst

```diff
@@ -63,8 +63,75 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
         exit 1;
     fi
 else
-    echo "macOS: Not implemented yet"
-    exit 1
+    # macOS
+
+    CONF_DIR="$INSTALL_DIR/etc"
+    APP_DIR="/Applications/Datadog Agent.app"
+
+    LOG_FILE="$LOG_DIR/preinstall.log"
+    exec > $LOG_FILE 2>&1
+
+    # Determine current user if he is using the Graphical installer
+    INSTALL_USER=$(ps aux | grep "CoreServices/Installer" | grep -v grep | awk '{print $1;}')
+
+    # Otherwise, we hope he is using the install script and try to use this user
+    # If it fails, no choice but to use root :'(
+    if [ -z "$INSTALL_USER" ] || [ "$INSTALL_USER" = "root" ]; then
+        SCRIPT_INSTALL="yes"
+        INSTALL_USER=`cat /tmp/datadog-install-user || echo 'root'`
+        rm -v /tmp/datadog-install-user || true
+    fi
+    echo "INSTALL_USER: $INSTALL_USER"
+
+    USER_HOME=`sudo -Hu $INSTALL_USER sh -c 'echo $HOME'`
+
+    if [ -e "$CONF_DIR/datadog.conf" ] || [ -e "$CONF_DIR/datadog.yaml" ]; then
+        echo "# State at the beginning"
+        echo "## Agent version"
+        grep AGENT_VERSION $INSTALL_DIR/agent/config.py || echo "No config.py file (agent 5)"
+        datadog-agent version || echo "No datadog-agent binary version (agent 6)"
+        echo "## $INSTALL_DIR"
+        ls -al $INSTALL_DIR || echo "No agent installed"
+        echo "## $APP_DIR/Contents/Resources"
+        ls -al "$APP_DIR/Contents/Resources" || echo "No app installed"
+
+        echo '# Stop old agent'
+        echo '## Trying to stop agent 5'
+        $INSTALL_DIR/bin/datadog-agent stop || true
+        echo '## Trying to stop agent 6'
+        sudo -Hu $INSTALL_USER launchctl stop com.datadoghq.agent || true
+        echo '## Trying to kill agent GUI'
+        kill `ps aux | grep 'Datadog Agent.app' | grep -v grep  | cut -d ' ' -f 4` || true
+
+        echo '## Trying to unload agent 6 launchd service'
```

Comment:
We don't stop the agent 5 launchd service (I believe it's `com.datadoghq.Agent.plist`)? Should we?

Reply:
`unload`ing the service should also stop it (at least that's what it does on my system :) )